src/components/AppProvider/utilities/createAppProviderContext/tests/createAppProviderContext.test.tsx

```diff
@@ -10,18 +11,19 @@ import Link from '../../Link';
 import {StickyManager} from '../../withSticky';
 import ScrollLockManager from '../../ScrollLockManager';
 
-jest.mock('@shopify/app-bridge');
-(createApp as jest.Mock<{}>).mockImplementation((args) => args);
-
 const actualIsServer = targets.isServer;
 
 function mockIsServer(value: boolean) {
   (targets as any).isServer = value;
 }
 
+let createAppSpy = jest.spyOn(appBridge, 'default');
```

Comment:
I think we should avoid calling `jest.spyOn` outside a test. We should do `jest.mock('@shopify/app-bridge')` outside of the tests and within the tests call `jest.spyOn`

Reply:
The reason behind the `jest.spyOn` outside of tests is that I wanted to keep using the same mock for most tests, but unmock the implementation for one of the tests using [`mockRestore()`](https://jestjs.io/docs/en/mock-function-api.html#mockfnmockrestore). Should I just repeat the mock for each test in this case? I've updated the test to do that instead, does this look better now?

Reply:
If you want the same mock for each test the preferred way is to use a factory function that is called in your test to set up the mocks rather than setting up a global. You can also just set up the mock in `beforeEach`. We also should avoid calling `jest.restoreAllMocks` since it's too general and explicitly call to restore individual mocks.

See the guide here: https://github.com/Shopify/web-foundation/blob/master/Best%20practices/Testing.md#setup-perform-assert